app/controllers/categories_controller.rb

```diff
@@ -0,0 +1,32 @@
+class CategoriesController < ApplicationController
+  def index
```

Comment:
- [ ] I am able to visit all your Category routes despite being logged out. But this should not be so. Kindly fix this.

  To do that you can add the following snippet to the top of your class:
  ```
  before_action :authenticate_user!
  ```

Reply:
Hi @TedLivist 

Thanks a lot for the good review. I would like to disagree. You can't access any categories or payments without being logged in. This is because I have already added the `before_action :authenticate_user!` helper in the `application controller` and it will be available to all other controllers since they all inherit from it. So no need to add that helper in the categories or payments controller(s).

Thanks!

Reply:
@JEFFKAY16 I don't think that's right. This is your `application_controller` as it is. There is no `authenticate_user!` helper there.

![jeff-helpers](https://user-images.githubusercontent.com/36542986/171906557-7491cd47-c142-4966-bb5d-10e5439e54be.jpg)

Reply:
@JEFFKAY16 I checked your app again. I am still able to visit pages when not logged in

Reply:
Hi @TedLivist 

Please see the picture below. It is a good practice to only add the helper in the application controller instead of duplicating it in all controllers. 
![Screenshot 2022-06-03 at 19 49 00](https://user-images.githubusercontent.com/84200026/171918543-15dfdd77-ef44-45da-93b9-9c021946339d.png)

Reply:
@TedLivist 

My bad, you were actually right. because I put the helper in the application controller, it was blocking my splash screen - so I hadt to put it in each controller. Thanks a lot mate!

Reply:
@JEFFKAY16 yeah. You're right about putting the helper in the application controller.

You're welcome.

Reply:
@JEFFKAY16 yeah. Nonetheless, you're right it's good practice to put the helper in the parent class instead of repeating it all over.

Reply:
@JEFFKAY16, You can add the helper `before_action :authenticate_user!` in the `application_controller` and then add this ` skip_before_action :authenticate_user!, only: [:index]` in the splash_controller so that the `authenticate_user!` helper will not have effect on the splash page and then you can remove the helper from other controllers so you have it only in the application_controller.

Reply:
Hi @julie-ify 
Thank you for raising this. I actually didn't know I can do that. If you can, please have it marked as optional, so that I can move forward but i will make sure to have that implemented. 

Thanks